### PR TITLE
feat: detailed rating dispaly

### PR DESCRIPTION
**Pull request by N-Feist**
[[AB#87500](https://dev.azure.com/plentymarkets/0af57df9-8662-4901-bb97-8e88b07ff0c9/_workitems/edit/87500)]

## Why:

Closes: #ID

## Checklist before requesting a review

- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

File: apps/web/components/ReviewsAccordion/ReviewsAccordion.vue [hunk 1]
```diff
@@ -9,14 +9,48 @@
...
+
+          <div class="flex flex-col">
+            <div v-for="(splitReview, key) in ratingPercentages" :key="key" class="flex items-center">
+              <p class="w-4 text-center">{{ 6 - key }}</p>
+              <SfRating class="mx-2 pb-1" size="base" :max="1" :value="1" />
+              <SfProgressLinear
+                class="self-center"
+                size="sm"
+                :value="splitReview"
+                aria-label="5star-Rating-Progress-Element"
```
Comment (Tim-M-S) [inline apps/web/components/ReviewsAccordion/ReviewsAccordion.vue]: Does this really describe what the element shows?

Comment (N-Feist) [reply apps/web/components/ReviewsAccordion/ReviewsAccordion.vue]: Not really, i already have it changed in an uncommitted change. Will push it soon, after i improved some parts of the new written code. @Tim-M-S

**Comment by csandru-plenty** [comment]
One more thing that I noticed @N-Feist , If you could have a look at the following, please (mevo system used):
http://localhost:3000/living-room/sofas/2-seater-red-carpet_135


Screenshot 2024-06-06 at 13 03 48
Screenshot 2024-06-06 at 13 04 00

**Comment by N-Feist** [comment]
This is caused, because i round the values as it says in the azure card: "Round down to the next half step. 4.3 will show 4 stars. 4.6 will show 4.5 stars". I have to round the value, to have the wanted value shown. I just havent applied this to every SFRating component, that already existed. 
Another option, would be to adjust the getters, to already have the ratings rounded. I just don't really like the approach @csandru-plenty

**Comment by csandru-plenty** [comment]
> This is caused, because i round the values as it says in the azure card: "Round down to the next half step. 4.3 will show 4 stars. 4.6 will show 4.5 stars". I have to round the value, to have the wanted value shown. I just havent applied this to every SFRating component, that already existed. Another option, would be to adjust the getters, to already have the ratings rounded. I just don't really like the approach @csandru-plenty

Ok, then if @Tim-M-S doesn't have anything against it, I'll approve. Please confirm.

**Comment by ksted** [comment]
> Another option, would be to adjust the getters, to already have the ratings rounded. I just don't really like the approach

Out of curiosity, what are your concerns with rounding in the getters?

**Comment by N-Feist** [comment]
Wouldn't we force the shop users then to have the ratings rounded, if we do it in the SDK? otherwise, it would be easy to change it up for our customers. Thats my main concerne, because i can image, some people wantting to display the full number. @ksted

File: apps/web/lang/en.json [hunk 2]
```diff
@@ -377,6 +377,8 @@
   "loadingReviews": "Loading Reviews...",
   "reviewsCount": "{count} reviews",
   "showAllReviews": "Show all reviews",
+  "avreageRating": "Average rating",
```
Comment (ksted) [inline apps/web/lang/en.json]:
```suggestion
  "averageRating": "Average rating",
```

**Comment by ksted** [comment]
> Wouldn't we force the shop users then to have the ratings rounded, if we do it in the SDK? otherwise, it would be easy to change it up for our customers. Thats my main concerne, because i can image, some people wantting to display the full number. @ksted

That's possible. But from a shopper's perspective, it would be pretty odd to see three different numbers across the page, like in the screenshot Cristi shared. I don't think this should happen in our defaults. I don't know enough about aria labels to say whether they should track the visual element (3.5 stars) or the real average (3.7) that's displayed next to the star rating, but I think we should try to be consistent.

Ideally it's also easy to be consistent here. So even if this kind of calculation happens inside the app instead of the getter, it should probably be centralised, so that both `PurchaseCard` and `ReviewsAccordion` use the same representation.

**Comment by N-Feist** [comment]
Ok so i will unify the way PurchaseCard and ReviewsAccordion get the review average. @ksted

**Comment by csandru-plenty** [comment]
Found a few issues while testing on mevo

eg: http://localhost:3000/living-room/sofas/2-seater-red-carpet_135

- hydration errors
- closing/opening CustomerReviews tab non-functional

Screenshot 2024-06-20 at 15 08 03
Screenshot 2024-06-20 at 15 08 39

**Review by csandru-plenty — APPROVED** [review]

**Pull request merged**